### DISCUSSION TITLE
Wiql generate

### DISCUSSION THREAD

**Discussion by kahaboom** [rfennell/AzurePipelines#1409] [Q&A] [answered]
Hello ! 
Is there a way to generate only query with a specific configurations with a **links** (for example Work item type - user stoty) ?

Thanks for answering :)

**Comment by rfennell** [comment, marked as answer]
Do you mean with the Release Notes task?

## WIQ 
You can find WI based on WIQ use the following parameters as detailed in the WIKI of this repo

| parameter | |
|-|-|
| wiqlWhereClause | An optional where clause to get a get a list of work items using a WIQL Query e.g. [System.TeamProject] = 'Project Name' and [System.WorkItemType] = 'Product Backlog Item'. The results of this query are available in the template in the queryWorkItems array. Note that this list of WI is independent of all other WI arrays.
|wiqlFromTarget | The FROM target for the WIQL Query. e.g. SELECT [System.Id] FROM workitems WHERE ... (Default is WorkItems)|

## Filters
The other option is to filter the WI found with the standard mechanism as discussed #1375 (it discusses filtering on tags, but same technique works for any field)

**Reply by kahaboom** [reply]
@rfennell Thank you for your reply !

Yes, Release Notes Tasks.

Currently I have the following setup : 
<img width="763" alt="Screenshot 2022-11-23 at 13 24 53" src="https://user-images.githubusercontent.com/80584072/203535132-0a502cdf-4dd0-423b-816c-e4bb2ee08ef1.png">
<img width="805" alt="Screenshot 2022-11-23 at 13 29 05" src="https://user-images.githubusercontent.com/80584072/203535879-68766be5-e785-4b5f-b294-a22944abda63.png">


And in the wiki I can see that it's uploading things that I want to see. But it's just like a text format without links, etc. So basically I can just look, but I cannot click on the workitem and go directly to it via link. Don't understand properly how am I able to add this feature.

**Reply by rfennell** [reply]
So, if I understand you correctly

- You have the expected list of WI being returned
- You need to make the list of WI be displayed as a HyperLink as opposed to simple text

You can do this in the handlebars template something like this

```
{{#forEach queryworkitems}}
*   - WI [{{this.id}}]({{replace this.url "_apis/wit/workItems" "_workitems/edit"}}) - {{lookup this.fields 'System.Title'}} 
{{/forEach}}
```

**Reply by kahaboom** [reply]
Sorry @rfennell for long answer. Your answer really helped me, thanks :)
